zerver/lib/actions.py

```diff
@@ -2389,6 +2403,20 @@ def do_update_message(user_profile, message_id, subject, propagate_mode, content
     else:
         raise JsonableError(_("You don't have permission to edit this message"))
 
+    # We already check for realm.allow_message_editing in
+    # zerver.views.messages.update_message_backend
+    # If there is a change to the content, we also need to check it hasn't been
+    # too long
+
+    # Allow an extra 20 seconds since we potentially allow editing 15 seconds
+    # past the limit, and in case there are network issues, etc. The 15 comes
+    # from (min_seconds_to_edit + seconds_left_buffer) in message_edit.js; if
+    # you change this value also change those two parameters in message_edit.js.
+    edit_limit_buffer = 20
+    if content is not None and user_profile.realm.message_content_edit_limit_seconds > 0 and \
+       (now() - message.pub_date) > datetime.timedelta(seconds=user_profile.realm.message_content_edit_limit_seconds + edit_limit_buffer):
+        raise JsonableError(_("The time limit for editing this message has past"))
```

Comment:
(You might just want to add another quick commit on top of this series moving the remaining JsonableError calls above; the only reason I didn't move those was to avoid merge-conflicting with your changes).